app/src/main/java/com/hynekbraun/openmeteoweather/presentation/mainscreen/WeatherViewModel.kt

```diff
@@ -0,0 +1,106 @@
+package com.hynekbraun.openmeteoweather.presentation.mainscreen
+
+import android.location.Location
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hynekbraun.openmeteoweather.domain.CurrentLocationManager
+import com.hynekbraun.openmeteoweather.domain.LocationError
+import com.hynekbraun.openmeteoweather.domain.WeatherFetchError
+import com.hynekbraun.openmeteoweather.domain.WeatherRepository
+import com.hynekbraun.openmeteoweather.domain.mapper.toCurrentData
+import com.hynekbraun.openmeteoweather.domain.mapper.toCurrentHourlyForecastData
+import com.hynekbraun.openmeteoweather.domain.mapper.toDailyForecastData
+import com.hynekbraun.openmeteoweather.domain.util.Resource
+import com.hynekbraun.openmeteoweather.presentation.mainscreen.util.*
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class WeatherViewModel @Inject constructor(
+    private val repository: WeatherRepository,
+    private val currentLocation: CurrentLocationManager
+) : ViewModel() {
+
+    var weatherState by mutableStateOf(WeatherState())
+        private set
+
+    private val eventChannel = Channel<ToastEventHandler>()
+    val eventFlow = eventChannel.receiveAsFlow()
+
+    fun onEvent(event: WeatherEvent) {
+        when (event) {
+            WeatherEvent.FetchData -> fetchData()
+        }
+    }
+
+    private fun fetchData() {
+        Log.d("TAG", "WeatherViewModel: fetch data triggered")
+        viewModelScope.launch {
+            val currentLocation = currentLocation.getLocation()
+            Log.d("TAG", "WeatherViewModel: Location fetched: $currentLocation")
+            when (currentLocation) {
+                is Resource.Error -> {
+                    fetchWeather(null)
```

Comment:
It was not there initially, but there is this error, when I turn the location on and start the app, I get Success from the Fused Location, but it is a null. I assume it is because the location is on but it is not loaded. So I did this as a workaround, because I don't know how to do it properly. 

<img width="1636" alt="Screenshot 2022-07-26 at 3 27 07" src="https://user-images.githubusercontent.com/84226147/180903356-1daebbae-a766-497b-9319-68f6b3398660.png">